### PR TITLE
http-proxy with new 'proxy-agent' package

### PR DESCRIPTION
**Pull request by panaC**
package : https://www.npmjs.com/package/proxy-agent
https://github.com/TooTallNate/proxy-agents/tree/main/packages/proxy-agent
source code : https://github.com/TooTallNate/proxy-agents/blob/main/packages/proxy-agent/src/index.ts

use env var with this package : https://www.npmjs.com/package/proxy-from-env

Tested with tinyproxy for mac https://github.com/tinyproxy/tinyproxy 

`brew install tinyproxy` 

```
cat << EOF > tinyproxy.conf   
Port 8888
Listen 127.0.0.1
Timeout 600
Allow 127.0.0.1
EOF
```

`tinyproxy -d -c tinyproxy.conf`

`http_proxy=127.0.0.1:8888 curl -v https://example.com`

In thorium when I choose the feedbooks opds feed with `HTTP_PROXY=http://127.0.0.1:8888 HTTPS_PROXY=http://127.0.0.1:8888 electron .`: 
```
[2]   proxy-agent Creating new ProxyAgent instance: { httpAgent: Agent { _events: [Object: null prototype] { free: [Function (anonymous)], newListener: [Function: maybeEnableKeylog] }, _eventsCount: 2, _maxListeners: undefined, defaultPort: 80, protocol: 'http:', options: [Object: null prototype] { timeout: 30000, noDelay: true, path: null }, requests: [Object: null prototype] {}, sockets: [Object: null prototype] {}, freeSockets: [Object: null prototype] {}, keepAliveMsecs: 1000, keepAlive: false, maxSockets: Infinity, maxFreeSockets: 256, scheduling: 'lifo', maxTotalSockets: Infinity, totalSocketCount: 0, [Symbol(kCapture)]: false }, httpsAgent: Agent { _events: [Object: null prototype] { free: [Function (anonymous)], newListener: [Function: maybeEnableKeylog] }, _eventsCount: 2, _maxListeners: undefined, defaultPort: 443, protocol: 'https:', options: [Object: null prototype] { timeout: 30000, rejectUnauthorized: false, noDelay: true, path: null }, requests: [Object: null prototype] {}, sockets: [Object: null prototype] {}, freeSockets: [Object: null prototype] {}, keepAliveMsecs: 1000, keepAlive: false, maxSockets: Infinity, maxFreeSockets: 256, scheduling: 'lifo', maxTotalSockets: Infinity, totalSocketCount: 0, maxCachedSessions: 100, _sessionCache: { map: {}, list: [] }, [Symbol(kCapture)]: false } } +8s
[2]   proxy-agent Request URL: 'https://catalog.feedbooks.com/catalog/index.json' +3ms
[2]   proxy-agent Proxy URL: 'http://127.0.0.1:8888' +0ms
[2]   readium-desktop:main/http fetch URL: https://catalog.feedbooks.com/catalog/index.json +8s
```

### CODE REVIEW

**Comment by panaC** [comment]
I hard coded the http env var in `package.json` for the test

**Comment by panaC** [comment]
https://github.com/TooTallNate/proxy-agents/blob/70023c12abe0d014004af6309ff7d0fdbaa60875/packages/proxy-agent/src/index.ts#L97-L99 

our custom http et https agent seems to be used

**Comment by danielweck** [comment]
TooTallNate's libs immediately appeared as a viable option, i am glad to see this working!!

https://github.com/edrlab/thorium-reader/issues/1904#issuecomment-1958065555

Thank you for this PR, we can remove the Electron version change and the hard-coded env vars, and we can merge?

File: package.json [hunk 1]
```diff
@@ -358,7 +359,7 @@
     "css-hot-loader": "^1.4.4",
     "css-loader": "^7.1.1",
     "devtron": "^1.4.0",
-    "electron": "^30.0.1",
```
Comment (danielweck) [inline package.json:361]: leave Electron version as-is

File: package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
     "start:dev:renderer:library": "webpack-dev-server --config webpack.config.renderer-library.js",
     "start:dev:renderer:reader": "webpack-dev-server --config webpack.config.renderer-reader.js",
     "start:dev:renderer:pdf": "webpack --config webpack.config.renderer-pdf.js",
-    "start:dev:main:electron": "cross-env DEBUG=r2:*,readium-desktop:*,-readium-desktop:main#streamerNoHttp,-readium-desktop:main:saga:persist,-r2:navigator#electron/common/readium-css-inject NODE_ENV=development electron .",
```
Comment (danielweck) [inline package.json:44]: remove test env vars

**Review by danielweck — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (panaC) [reply package.json:44]: done

File: package.json (hunk 1 above)
Comment (panaC) [reply package.json]: done

**Pull request merged**